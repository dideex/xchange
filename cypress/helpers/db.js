const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
