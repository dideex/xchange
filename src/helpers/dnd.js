import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
