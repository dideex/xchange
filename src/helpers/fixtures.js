export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}