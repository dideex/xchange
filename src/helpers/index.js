export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
