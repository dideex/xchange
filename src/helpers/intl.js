import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
