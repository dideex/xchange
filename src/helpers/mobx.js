import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
