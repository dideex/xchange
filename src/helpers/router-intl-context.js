import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
