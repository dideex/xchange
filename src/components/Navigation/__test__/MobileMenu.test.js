import React from 'react'
import {Provider as MobxProvider} from 'mobx-react'

import Component from '../MobileMenu'
import UserStore from '../../../store/User'
import {fakeUser} from '../../../helpers/fixtures'
import {mountWrap} from '../../../helpers/router-intl-context'

const NO_OP = () => {}
describe('Navigation: lang menu', () => {
  const props = {
    handleClick: NO_OP,
    showLangMenu: false,
    showMobMenu: false,
    closeBothMenu: NO_OP,
  }
  let userStore = new UserStore()
  beforeEach(() => {
    userStore = new UserStore()
  })
  describe('Markup', () => {
    it('Basic markup', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component  {...props} />
        </MobxProvider>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Should render lang menu', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component  {...props} showLangMenu />
        </MobxProvider>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Should render mob menu', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component  {...props} showMobMenu />
        </MobxProvider>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('Component behavriour', () => {
    it('Handle click should invoke on clicking', () => {
      const handleClick = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component  {...props} handleClick={handleClick} />
        </MobxProvider>,
      )
      wrapper.find('[data-testid="mobile-menu-lang"]').simulate('click')

      expect(handleClick).toHaveBeenCalledTimes(1)
    })

    it('Close menu should invoke on clicking', () => {
      const closeBothMenu = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component  {...props} closeBothMenu={closeBothMenu} />
        </MobxProvider>,
      )
      wrapper.find('#close').simulate('click')

      expect(closeBothMenu).toHaveBeenCalledTimes(1)
    })
  })
})
