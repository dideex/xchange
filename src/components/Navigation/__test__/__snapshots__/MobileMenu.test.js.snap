// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation: lang menu Markup Basic markup 1`] = `"<div x=\\"100%\\" class=\\"css-iuahf9\\"><div class=\\"close\\"><svg style=\\"width: 30px;\\" class=\\"\\" viewBox=\\"0 0 35 34\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><rect x=\\"0.882812\\" y=\\"29.3744\\" width=\\"41\\" height=\\"6\\" transform=\\"rotate(-45 0.882812 29.3744)\\" fill=\\"white\\"></rect><path d=\\"M5.12598 0.382996L34.1174 29.3744L29.8747 33.617L0.883336 4.62564L5.12598 0.382996Z\\" fill=\\"white\\"></path></svg></div><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"mobile-menu-lang\\">Язык</span><a href=\\"/lichnii-kabinet\\"><span>Signin</span></a><a href=\\"/registracya\\"><span>Signup</span></a></div>"`;

exports[`Navigation: lang menu Markup Should render lang menu 1`] = `"<div x=\\"100%\\" class=\\"css-iuahf9\\"><div class=\\"close\\"><svg style=\\"width: 30px;\\" class=\\"\\" viewBox=\\"0 0 35 34\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><rect x=\\"0.882812\\" y=\\"29.3744\\" width=\\"41\\" height=\\"6\\" transform=\\"rotate(-45 0.882812 29.3744)\\" fill=\\"white\\"></rect><path d=\\"M5.12598 0.382996L34.1174 29.3744L29.8747 33.617L0.883336 4.62564L5.12598 0.382996Z\\" fill=\\"white\\"></path></svg></div><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"mobile-menu-lang\\">Язык</span><p><span data-testid=\\"ru-RU\\">Рус</span><span data-testid=\\"en-US\\">Eng</span><span data-testid=\\"fr-FR\\">Fr</span><span data-testid=\\"zh-ZH\\">Cn</span></p><a href=\\"/lichnii-kabinet\\"><span>Signin</span></a><a href=\\"/registracya\\"><span>Signup</span></a></div>"`;

exports[`Navigation: lang menu Markup Should render mob menu 1`] = `"<div x=\\"0%\\" class=\\"css-1ew0030\\"><div class=\\"close\\"><svg style=\\"width: 30px;\\" class=\\"\\" viewBox=\\"0 0 35 34\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><rect x=\\"0.882812\\" y=\\"29.3744\\" width=\\"41\\" height=\\"6\\" transform=\\"rotate(-45 0.882812 29.3744)\\" fill=\\"white\\"></rect><path d=\\"M5.12598 0.382996L34.1174 29.3744L29.8747 33.617L0.883336 4.62564L5.12598 0.382996Z\\" fill=\\"white\\"></path></svg></div><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"mobile-menu-lang\\">Язык</span><a href=\\"/lichnii-kabinet\\"><span>Signin</span></a><a href=\\"/registracya\\"><span>Signup</span></a></div>"`;
